src/commands/projects.ts

```diff
@@ -172,14 +192,34 @@ const deleteProject = async (props: CommonProps & IdOrNameProps) => {
 const update = async (
   props: CommonProps &
     IdOrNameProps & {
-      name: string;
+      name?: string;
+      ipAllow?: string[];
+      ipPrimaryOnly?: boolean;
     },
 ) => {
+  const project: ProjectUpdateRequest['project'] = {};
+  if (props.name) {
+    project.name = props.name;
+  }
+  if (props.ipAllow || props.ipPrimaryOnly != undefined) {
+    const { data } = await props.apiClient.getProject(props.id);
+    const existingAllowedIps = data.project.settings?.allowed_ips;
+
+    project.settings = {
+      allowed_ips: {
+        ips: props.ipAllow ?? existingAllowedIps?.ips ?? [],
+        primary_branch_only:
+          props.ipPrimaryOnly ??
+          existingAllowedIps?.primary_branch_only ??
+          false,
+      },
+    };
+  }
+
   const { data } = await props.apiClient.updateProject(props.id, {
-    project: {
-      name: props.name,
-    },
+    project,
   });
+
   writer(props).end(data.project, { fields: PROJECT_FIELDS });
```

Comment:
It is displayed for `json` and `yaml` output formats, For the `table` format, we have to show the entire `settings` object (`allowed_ips` + `quota` as a single column) if a user updates the IP allowed. I can try displaying it as stringified JSON, but it might look weirdly formatted.

Reply:
hmm, let me see

Reply:
yea, it will look ugly. We either should bring custom formatted either leave it as it is

Reply:
I can do something like this, but with `settings.quota` not sure how that last column will look like.

<img width="1655" alt="ip-allow-table-settings" src="https://github.com/neondatabase/neonctl/assets/1949852/6fd3614d-4ff8-4bd2-b8bf-56c5310dd4bf">

Reply:
I'd rather display something custom like:
```
ID                 |  Name          | ... | Allowed IPs
----------------------------------------------------------------------------
lalala-123    | my project   | ... |  IPs: 123.0.0.1, 1.1.1.1 - 2.2.2.2, ...
                                      Primary branch only: true
```

Reply:
I simply used `YAML.stringify` to quickly generate the above formatting. For your suggestion, I need to implement our own `stringify` method. Also,  need to check how that format will look on smaller screens. 

Also, another challenge is the `writer#end` does not support nested keys (e.g. `settings.allowed_ips`), and needs to play with the `WriterOutConfig` type to accept nested fields. Maybe I can work on that as a separate PR, if we still need to.

Reply:
Yea, as I said, let's either do it pretty, or skip :)